src/AutoRest.CSharp/Common/Output/Models/Shared/Parameter.cs

```diff
@@ -131,7 +131,8 @@ private static string ConstructParameterVariableName(InputParameter param, CShar
             {
                 var paramInputTypeName = paramInputType.Name;
 
-                if (paramName.Equals(paramInputTypeName))
+                if (paramName.Equals(paramInputTypeName) || // remove this after adoption of TCGC
```

Comment:
will this PR merge after the TCGC adoption?

Reply:
> will this PR merge after the TCGC adoption?

It will merge before TCGC adoption. After TCGC adoption, all model names should start with lower case, so the `Equals()` will be deleted.